File: examples/AdditionServerComponent_150/src/urn_jaus_example_addition_server_1_0/Messages/QueryAddition.cpp
#include "urn_jaus_example_addition_server_1_0/Messages/QueryAddition.h"

namespace urn_jaus_example_addition_server_1_0
{

void QueryAddition::Header::HeaderRecord::setParent(Header* parent)
{
	m_parent = parent;
}

void QueryAddition::Header::HeaderRecord::setParentPresenceVector()
{
	if(m_parent != NULL )
	{
		m_parent->setParentPresenceVector();
	}
}

jUnsignedShortInteger QueryAddition::Header::HeaderRecord::getMessageIDHeader()
{
	return m_MessageIDHeader;
}

int QueryAddition::Header::HeaderRecord::setMessageIDHeader(jUnsignedShortInteger value)
{
	m_MessageIDHeader = value;
	setParentPresenceVector();
	return 0;
}

/**
 * Returns the size of memory the used data members of the class occupies.
 * This is not necessarily the same size of memory the class actually occupies.
 * Eg. A union of an int and a double may occupy 8 bytes. However, if the data
 *     stored is an int, this function will return a size of 4 bytes.
 * 
 * @return
 */
const unsigned int QueryAddition::Header::HeaderRecord::getSize()
{
	unsigned int size = 0;
	
	size += sizeof(jUnsignedShortInteger);
	
	return size;
}

void QueryAddition::Header::HeaderRecord::encode(unsigned char *bytes)
{
	
	if (bytes == NULL)
	{
		return;
	}
	
	int pos = 0;
	jUnsignedShortInteger m_MessageIDHeaderTemp;
	
	m_MessageIDHeaderTemp = JSIDL_v_1_0::correctEndianness(m_MessageIDHeader);
	memcpy(bytes + pos, &m_MessageIDHeaderTemp, sizeof(jUnsignedShortInteger));
	pos += sizeof(jUnsignedShortInteger);
}

void QueryAddition::Header::HeaderRecord::decode(const unsigned char *bytes)
{
	
	if (bytes == NULL)
	{
		return;
	}
	
	int pos = 0;
	jUnsignedShortInteger m_MessageIDHeaderTemp;
	
	memcpy(&m_MessageIDHeaderTemp, bytes + pos, sizeof(jUnsignedShortInteger));
	m_MessageIDHeader = JSIDL_v_1_0::correctEndianness(m_MessageIDHeaderTemp);
	pos += sizeof(jUnsignedShortInteger);
}

QueryAddition::Header::HeaderRecord &QueryAddition::Header::HeaderRecord::operator=(const HeaderRecord &value)
{
	m_MessageIDHeader = value.m_MessageIDHeader;
	
	return *this;
}

bool QueryAddition::Header::HeaderRecord::operator==(const HeaderRecord &value) const
{
	if (m_MessageIDHeader != value.m_MessageIDHeader)
	{
		return false;
	}
	
	return true;
}

bool QueryAddition::Header::HeaderRecord::operator!=(const HeaderRecord &value) const
{
	return !((*this) == value);
}

QueryAddition::Header::HeaderRecord::HeaderRecord()
{
	m_parent = NULL;
	m_MessageIDHeader = 0xf010;
}

QueryAddition::Header::HeaderRecord::HeaderRecord(const HeaderRecord &value)
{
	/// Initiliaze the protected variables
	m_parent = NULL;
	m_MessageIDHeader = 0xf010;
	
	/// Copy the values
	m_MessageIDHeader = value.m_MessageIDHeader;
}

QueryAddition::Header::HeaderRecord::~HeaderRecord()
{
}

QueryAddition::Header::HeaderRecord* const QueryAddition::Header::getHeaderRecord()
{
	return &m_HeaderRecord;
}

int QueryAddition::Header::setHeaderRecord(const HeaderRecord &value)
{
	m_HeaderRecord = value;
	setParentPresenceVector();
	return 0;
}

void QueryAddition::Header::setParentPresenceVector()
{
	// Nothing needed here, placeholder function
}

/**
 * Returns the size of memory the used data members of the class occupies.
 * This is not necessarily the same size of memory the class actually occupies.
 * Eg. A union of an int and a double may occupy 8 bytes. However, if the data
 *     stored is an int, this function will return a size of 4 bytes.
 * 
 * @return
 */
const unsigned int QueryAddition::Header::getSize()
{
	unsigned int size = 0;
	
	size += m_HeaderRecord.getSize();
	
	return size;
}

void QueryAddition::Header::encode(unsigned char *bytes)
{
	
	if (bytes == NULL)
	{
		return;
	}
	
	int pos = 0;
	m_HeaderRecord.encode(bytes + pos);
	pos += m_HeaderRecord.getSize();
}

void QueryAddition::Header::decode(const unsigned char *bytes)
{
	
	if (bytes == NULL)
	{
		return;
	}
	
	int pos = 0;
	m_HeaderRecord.decode(bytes + pos);
	pos += m_HeaderRecord.getSize();
}

QueryAddition::Header &QueryAddition::Header::operator=(const Header &value)
{
	m_HeaderRecord = value.m_HeaderRecord;
	m_HeaderRecord.setParent(this);
	
	return *this;
}

bool QueryAddition::Header::operator==(const Header &value) const
{
	if (m_HeaderRecord != value.m_HeaderRecord)
	{
		return false;
	}
	return true;
}

bool QueryAddition::Header::operator!=(const Header &value) const
{
	return !((*this) == value);
}

QueryAddition::Header::Header()
{
	m_HeaderRecord.setParent(this);
	/// No Initialization of m_HeaderRecord necessary.
}

QueryAddition::Header::Header(const Header &value)
{
	/// Initiliaze the protected variables
	m_HeaderRecord.setParent(this);
	/// No Initialization of m_HeaderRecord necessary.
	
	/// Copy the values
	m_HeaderRecord = value.m_HeaderRecord;
	m_HeaderRecord.setParent(this);
}

QueryAddition::Header::~Header()
{
}

QueryAddition::Header* const QueryAddition::getHeader()
{
	return &m_Header;
}

int QueryAddition::setHeader(const Header &value)
{
	m_Header = value;
	return 0;
}

void QueryAddition::AdditionInputBody::AdditionInput::setParent(AdditionInputBody* parent)
{
	m_parent = parent;
}

void QueryAddition::AdditionInputBody::AdditionInput::setParentPresenceVector()
{
	if(m_parent != NULL )
	{
		m_parent->setParentPresenceVector();
	}
}

jUnsignedInteger QueryAddition::AdditionInputBody::AdditionInput::getA1()
{
	return m_A1;
}

int QueryAddition::AdditionInputBody::AdditionInput::setA1(jUnsignedInteger value)
{
	m_A1 = value;
	setParentPresenceVector();
	return 0;
}

jUnsignedInteger QueryAddition::AdditionInputBody::AdditionInput::getA2()
{
	return m_A2;
}

int QueryAddition::AdditionInputBody::AdditionInput::setA2(jUnsignedInteger value)
{
	m_A2 = value;
	setParentPresenceVector();
	return 0;
}

/**
 * Returns the size of memory the used data members of the class occupies.
 * This is not necessarily the same size of memory the class actually occupies.
 * Eg. A union of an int and a double may occupy 8 bytes. However, if the data
 *     stored is an int, this function will return a size of 4 bytes.
 * 
 * @return
 */
const unsigned int QueryAddition::AdditionInputBody::AdditionInput::getSize()
{
	unsigned int size = 0;
	
	size += sizeof(jUnsignedInteger);
	size += sizeof(jUnsignedInteger);
	
	return size;
}

void QueryAddition::AdditionInputBody::AdditionInput::encode(unsigned char *bytes)
{
	
	if (bytes == NULL)
	{
		return;
	}
	
	int pos = 0;
	jUnsignedInteger m_A1Temp;
	
	m_A1Temp = JSIDL_v_1_0::correctEndianness(m_A1);
	memcpy(bytes + pos, &m_A1Temp, sizeof(jUnsignedInteger));
	pos += sizeof(jUnsignedInteger);
	jUnsignedInteger m_A2Temp;
	
	m_A2Temp = JSIDL_v_1_0::correctEndianness(m_A2);
	memcpy(bytes + pos, &m_A2Temp, sizeof(jUnsignedInteger));
	pos += sizeof(jUnsignedInteger);
}

void QueryAddition::AdditionInputBody::AdditionInput::decode(const unsigned char *bytes)
{
	
	if (bytes == NULL)
	{
		return;
	}
	
	int pos = 0;
	jUnsignedInteger m_A1Temp;
	
	memcpy(&m_A1Temp, bytes + pos, sizeof(jUnsignedInteger));
	m_A1 = JSIDL_v_1_0::correctEndianness(m_A1Temp);
	pos += sizeof(jUnsignedInteger);
	jUnsignedInteger m_A2Temp;
	
	memcpy(&m_A2Temp, bytes + pos, sizeof(jUnsignedInteger));
	m_A2 = JSIDL_v_1_0::correctEndianness(m_A2Temp);
	pos += sizeof(jUnsignedInteger);
}

QueryAddition::AdditionInputBody::AdditionInput &QueryAddition::AdditionInputBody::AdditionInput::operator=(const AdditionInput &value)
{
	m_A1 = value.m_A1;
	m_A2 = value.m_A2;
	
	return *this;
}

bool QueryAddition::AdditionInputBody::AdditionInput::operator==(const AdditionInput &value) const
{
	if (m_A1 != value.m_A1)
	{
		return false;
	}
	if (m_A2 != value.m_A2)
	{
		return false;
	}
	
	return true;
}

bool QueryAddition::AdditionInputBody::AdditionInput::operator!=(const AdditionInput &value) const
{
	return !((*this) == value);
}

QueryAddition::AdditionInputBody::AdditionInput::AdditionInput()
{
	m_parent = NULL;
	m_A1 = 0;
	m_A2 = 0;
}

QueryAddition::AdditionInputBody::AdditionInput::AdditionInput(const AdditionInput &value)
{
	/// Initiliaze the protected variables
	m_parent = NULL;
	m_A1 = 0;
	m_A2 = 0;
	
	/// Copy the values
	m_A1 = value.m_A1;
	m_A2 = value.m_A2;
}

QueryAddition::AdditionInputBody::AdditionInput::~AdditionInput()
{
}

QueryAddition::AdditionInputBody::AdditionInput* const QueryAddition::AdditionInputBody::getAdditionInput()
{
	return &m_AdditionInput;
}

int QueryAddition::AdditionInputBody::setAdditionInput(const AdditionInput &value)
{
	m_AdditionInput = value;
	setParentPresenceVector();
	return 0;
}

void QueryAddition::AdditionInputBody::setParentPresenceVector()
{
	// Nothing needed here, placeholder function
}

/**
 * Returns the size of memory the used data members of the class occupies.
 * This is not necessarily the same size of memory the class actually occupies.
 * Eg. A union of an int and a double may occupy 8 bytes. However, if the data
 *     stored is an int, this function will return a size of 4 bytes.
 * 
 * @return
 */
const unsigned int QueryAddition::AdditionInputBody::getSize()
{
	unsigned int size = 0;
	
	size += m_AdditionInput.getSize();
	
	return size;
}

void QueryAddition::AdditionInputBody::encode(unsigned char *bytes)
{
	
	if (bytes == NULL)
	{
		return;
	}
	
	int pos = 0;
	m_AdditionInput.encode(bytes + pos);
	pos += m_AdditionInput.getSize();
}

void QueryAddition::AdditionInputBody::decode(const unsigned char *bytes)
{
	
	if (bytes == NULL)
	{
		return;
	}
	
	int pos = 0;
	m_AdditionInput.decode(bytes + pos);
	pos += m_AdditionInput.getSize();
}

QueryAddition::AdditionInputBody &QueryAddition::AdditionInputBody::operator=(const AdditionInputBody &value)
{
	m_AdditionInput = value.m_AdditionInput;
	m_AdditionInput.setParent(this);
	/// This code is currently not supported
	
	return *this;
}

bool QueryAddition::AdditionInputBody::operator==(const AdditionInputBody &value) const
{
	if (m_AdditionInput != value.m_AdditionInput)
	{
		return false;
	}
	/// This code is currently not supported
	return true;
}

bool QueryAddition::AdditionInputBody::operator!=(const AdditionInputBody &value) const
{
	return !((*this) == value);
}

QueryAddition::AdditionInputBody::AdditionInputBody()
{
	m_AdditionInput.setParent(this);
	/// No Initialization of m_AdditionInput necessary.
}

QueryAddition::AdditionInputBody::AdditionInputBody(const AdditionInputBody &value)
{
	/// Initiliaze the protected variables
	m_AdditionInput.setParent(this);
	/// No Initialization of m_AdditionInput necessary.
	
	/// Copy the values
	m_AdditionInput = value.m_AdditionInput;
	m_AdditionInput.setParent(this);
	/// This code is currently not supported
}

QueryAddition::AdditionInputBody::~AdditionInputBody()
{
}

QueryAddition::AdditionInputBody* const QueryAddition::getAdditionInputBody()
{
	return &m_AdditionInputBody;
}

int QueryAddition::setAdditionInputBody(const AdditionInputBody &value)
{
	m_AdditionInputBody = value;
	return 0;
}

/**
 * Returns the size of memory the used data members of the class occupies.
 * This is not necessarily the same size of memory the class actually occupies.
 * Eg. A union of an int and a double may occupy 8 bytes. However, if the data
 *     stored is an int, this function will return a size of 4 bytes.
 * 
 * @return
 */
const unsigned int QueryAddition::getSize()
{
	unsigned int size = 0;
	
	size += m_Header.getSize();
	size += m_AdditionInputBody.getSize();
	
	return size;
}

void QueryAddition::encode(unsigned char *bytes)
{
	
	if (bytes == NULL)
	{
		return;
	}
	
	int pos = 0;
	m_Header.encode(bytes + pos);
	pos += m_Header.getSize();
	m_AdditionInputBody.encode(bytes + pos);
	pos += m_AdditionInputBody.getSize();
}

void QueryAddition::decode(const unsigned char *bytes)
{
	
	if (bytes == NULL)
	{
		return;
	}
	
	int pos = 0;
	m_Header.decode(bytes + pos);
	pos += m_Header.getSize();
	m_AdditionInputBody.decode(bytes + pos);
	pos += m_AdditionInputBody.getSize();
}

QueryAddition &QueryAddition::operator=(const QueryAddition &value)
{
	m_Header = value.m_Header;
	m_AdditionInputBody = value.m_AdditionInputBody;
	
	return *this;
}

bool QueryAddition::operator==(const QueryAddition &value) const
{
	if (m_Header != value.m_Header)
	{
		return false;
	}
	if (m_AdditionInputBody != value.m_AdditionInputBody)
	{
		return false;
	}
	
	return true;
}

bool QueryAddition::operator!=(const QueryAddition &value) const
{
	return !((*this) == value);
}

QueryAddition::QueryAddition()
{
	/// No Initialization of m_Header necessary.
	/// No Initialization of m_AdditionInputBody necessary.
	m_IsCommand = false;
}

QueryAddition::QueryAddition(const QueryAddition &value)
{
	/// Initiliaze the protected variables
	/// No Initialization of m_Header necessary.
	/// No Initialization of m_AdditionInputBody necessary.
	m_IsCommand = false;
	
	/// Copy the values
	m_Header = value.m_Header;
	m_AdditionInputBody = value.m_AdditionInputBody;
}

QueryAddition::~QueryAddition()
{
}


}

File: examples/Java/AdditionServerComponent_150/src/urn_jaus_example_addition_server_1_0/Messages/QueryAddition.java
package src.urn_jaus_example_addition_server_1_0.Messages;

import framework.messages.Message;
import framework.JausUtils;
import java.nio.ByteBuffer;
import java.nio.ByteOrder;
import java.util.BitSet;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.logging.Level;
import java.util.logging.Logger;

public class QueryAddition extends Message
{
	protected static Logger logger = Logger.getLogger("framework.logger");
	public static int ID = 0xf010;
	
	public static class  Header
	{
		public static class  HeaderRecord
		{
		
			protected Header m_parent;
			protected int m_MessageIDHeader;
		
			public void setParent(Header parent)
			{
				m_parent = parent;
			}
			
			public void setParentPresenceVector()
			{
				if(m_parent != null )
				{
					m_parent.setParentPresenceVector();
				}
			}
			
			public int getMessageIDHeader()
			{
				return m_MessageIDHeader;
			}
			
			public void setMessageIDHeader(int value)
			{
				m_MessageIDHeader = value;
				setParentPresenceVector();
			}
			
			/**
			 * Returns the number of bytes the used data members of the class occupies
			 * in the buffer. This is not the number of bytes the data type occupies in 
			 * Java, but the bytes expected on the wire.
			 * 
			 * @return
			 */
			public long getSize()
			{
				long size = 0;
				
				size += JausUtils.getNumBytes("short");
				
				return size;
			}
			
			public void encode(ByteBuffer bytes, int pos)
			{
				
				if (bytes.array() == null)
				{
					return;
				}
				
				if(bytes.order() != ByteOrder.LITTLE_ENDIAN)
				{
					bytes.order(ByteOrder.LITTLE_ENDIAN);
				}
				
				bytes.putShort(pos, (short) m_MessageIDHeader);
				pos += JausUtils.getNumBytes("short");
			}
			
			public void decode(ByteBuffer bytes, int pos)
			{
				
				if (bytes.array() == null)
				{
					return;
				}
				if(bytes.order() != ByteOrder.LITTLE_ENDIAN)
				{
					bytes.order(ByteOrder.LITTLE_ENDIAN);
				}
				
				m_MessageIDHeader = bytes.getShort(pos) & 0xffff;
				pos += JausUtils.getNumBytes("short");
			}
			
			public QueryAddition.Header.HeaderRecord assign(HeaderRecord value)
			{
				m_MessageIDHeader = value.m_MessageIDHeader;
				
				return this;
			}
			
			public boolean isEqual(HeaderRecord value)
			{
				if (m_MessageIDHeader != value.getMessageIDHeader())
				{
					return false;
				}
				
				return true;
			}
			
			public boolean notEquals(HeaderRecord value)
			{
				return !this.isEqual(value);
				}
				
			public HeaderRecord()
			{
				m_parent = null;
				m_MessageIDHeader = 0xf010;
			}
			
			public HeaderRecord(HeaderRecord value)
			{
				/// Initiliaze the protected variables
				m_parent = null;
				m_MessageIDHeader = 0xf010;
				
				/// Copy the values
				m_MessageIDHeader = value.m_MessageIDHeader;
			}
			
			public void finalize()
			{
			}
			
		}
	
	
		protected HeaderRecord m_HeaderRecord;
	
		public QueryAddition.Header.HeaderRecord getHeaderRecord()
		{
			return m_HeaderRecord;
		}
		
		public void setHeaderRecord(HeaderRecord value)
		{
			m_HeaderRecord = value;
			setParentPresenceVector();
		}
		
		public void setParentPresenceVector()
		{
			// Nothing needed here, placeholder function
		}
		
		/**
		 * Returns the number of bytes the used data members of the class occupies
		 * in the buffer. This is not the number of bytes the data type occupies in 
		 * Java, but the bytes expected on the wire.
		 * 
		 * @return
		 */
		public long getSize()
		{
			long size = 0;
			
			size += m_HeaderRecord.getSize();
			
			return size;
		}
		
		public void encode(ByteBuffer bytes, int pos)
		{
			
			if (bytes.array() == null)
			{
				return;
			}
			
			if(bytes.order() != ByteOrder.LITTLE_ENDIAN)
			{
				bytes.order(ByteOrder.LITTLE_ENDIAN);
			}
			
			m_HeaderRecord.encode(bytes, pos);
			pos += m_HeaderRecord.getSize();
		}
		
		public void decode(ByteBuffer bytes, int pos)
		{
			
			if (bytes.array() == null)
			{
				return;
			}
			if(bytes.order() != ByteOrder.LITTLE_ENDIAN)
			{
				bytes.order(ByteOrder.LITTLE_ENDIAN);
			}
			
			m_HeaderRecord.decode(bytes, pos);
			pos += m_HeaderRecord.getSize();
		}
		
		public QueryAddition.Header assign(Header value)
		{
			m_HeaderRecord = value.m_HeaderRecord;
			m_HeaderRecord.setParent(this);
			
			return this;
		}
		
		public boolean isEqual(Header value)
		{
			if (!m_HeaderRecord.isEqual(value.getHeaderRecord()))
			{
				return false;
			}
			return true;
		}
		
		public boolean notEquals(Header value)
		{
			return !this.isEqual(value);
			}
		
		public Header()
		{
			m_HeaderRecord = new HeaderRecord();
			m_HeaderRecord.setParent(this);
		}
		
		public Header(Header value)
		{
			/// Initiliaze the protected variables
			m_HeaderRecord = new HeaderRecord();
			m_HeaderRecord.setParent(this);
			
			/// Copy the values
			m_HeaderRecord = value.m_HeaderRecord;
			m_HeaderRecord.setParent(this);
		}
		
		public void finalize()
		{
		}
		
	}
	public static class  AdditionInputBody
	{
		public static class  AdditionInput
		{
		
			protected AdditionInputBody m_parent;
			protected long m_A1;
			protected long m_A2;
		
			public void setParent(AdditionInputBody parent)
			{
				m_parent = parent;
			}
			
			public void setParentPresenceVector()
			{
				if(m_parent != null )
				{
					m_parent.setParentPresenceVector();
				}
			}
			
			public long getA1()
			{
				return m_A1;
			}
			
			public void setA1(long value)
			{
				m_A1 = value;
				setParentPresenceVector();
			}
			
			public long getA2()
			{
				return m_A2;
			}
			
			public void setA2(long value)
			{
				m_A2 = value;
				setParentPresenceVector();
			}
			
			/**
			 * Returns the number of bytes the used data members of the class occupies
			 * in the buffer. This is not the number of bytes the data type occupies in 
			 * Java, but the bytes expected on the wire.
			 * 
			 * @return
			 */
			public long getSize()
			{
				long size = 0;
				
				size += JausUtils.getNumBytes("int");
				size += JausUtils.getNumBytes("int");
				
				return size;
			}
			
			public void encode(ByteBuffer bytes, int pos)
			{
				
				if (bytes.array() == null)
				{
					return;
				}
				
				if(bytes.order() != ByteOrder.LITTLE_ENDIAN)
				{
					bytes.order(ByteOrder.LITTLE_ENDIAN);
				}
				
				bytes.putInt(pos, (int) m_A1);
				pos += JausUtils.getNumBytes("int");
				bytes.putInt(pos, (int) m_A2);
				pos += JausUtils.getNumBytes("int");
			}
			
			public void decode(ByteBuffer bytes, int pos)
			{
				
				if (bytes.array() == null)
				{
					return;
				}
				if(bytes.order() != ByteOrder.LITTLE_ENDIAN)
				{
					bytes.order(ByteOrder.LITTLE_ENDIAN);
				}
				
				m_A1 = bytes.getInt(pos) & 0xffffffffL;
				pos += JausUtils.getNumBytes("int");
				m_A2 = bytes.getInt(pos) & 0xffffffffL;
				pos += JausUtils.getNumBytes("int");
			}
			
			public QueryAddition.AdditionInputBody.AdditionInput assign(AdditionInput value)
			{
				m_A1 = value.m_A1;
				m_A2 = value.m_A2;
				
				return this;
			}
			
			public boolean isEqual(AdditionInput value)
			{
				if (m_A1 != value.getA1())
				{
					return false;
				}
				if (m_A2 != value.getA2())
				{
					return false;
				}
				
				return true;
			}
			
			public boolean notEquals(AdditionInput value)
			{
				return !this.isEqual(value);
				}
				
			public AdditionInput()
			{
				m_parent = null;
				m_A1 = 0;
				m_A2 = 0;
			}
			
			public AdditionInput(AdditionInput value)
			{
				/// Initiliaze the protected variables
				m_parent = null;
				m_A1 = 0;
				m_A2 = 0;
				
				/// Copy the values
				m_A1 = value.m_A1;
				m_A2 = value.m_A2;
			}
			
			public void finalize()
			{
			}
			
		}
	
	
		protected AdditionInput m_AdditionInput;
	
		public QueryAddition.AdditionInputBody.AdditionInput getAdditionInput()
		{
			return m_AdditionInput;
		}
		
		public void setAdditionInput(AdditionInput value)
		{
			m_AdditionInput = value;
			setParentPresenceVector();
		}
		
		public void setParentPresenceVector()
		{
			// Nothing needed here, placeholder function
		}
		
		/**
		 * Returns the number of bytes the used data members of the class occupies
		 * in the buffer. This is not the number of bytes the data type occupies in 
		 * Java, but the bytes expected on the wire.
		 * 
		 * @return
		 */
		public long getSize()
		{
			long size = 0;
			
			size += m_AdditionInput.getSize();
			
			return size;
		}
		
		public void encode(ByteBuffer bytes, int pos)
		{
			
			if (bytes.array() == null)
			{
				return;
			}
			
			if(bytes.order() != ByteOrder.LITTLE_ENDIAN)
			{
				bytes.order(ByteOrder.LITTLE_ENDIAN);
			}
			
			m_AdditionInput.encode(bytes, pos);
			pos += m_AdditionInput.getSize();
		}
		
		public void decode(ByteBuffer bytes, int pos)
		{
			
			if (bytes.array() == null)
			{
				return;
			}
			if(bytes.order() != ByteOrder.LITTLE_ENDIAN)
			{
				bytes.order(ByteOrder.LITTLE_ENDIAN);
			}
			
			m_AdditionInput.decode(bytes, pos);
			pos += m_AdditionInput.getSize();
		}
		
		public QueryAddition.AdditionInputBody assign(AdditionInputBody value)
		{
			m_AdditionInput = value.m_AdditionInput;
			m_AdditionInput.setParent(this);
			/// This code is currently not supported
			
			return this;
		}
		
		public boolean isEqual(AdditionInputBody value)
		{
			if (!m_AdditionInput.isEqual(value.getAdditionInput()))
			{
				return false;
			}
			/// This code is currently not supported
			return true;
		}
		
		public boolean notEquals(AdditionInputBody value)
		{
			return !this.isEqual(value);
			}
		
		public AdditionInputBody()
		{
			m_AdditionInput = new AdditionInput();
			m_AdditionInput.setParent(this);
		}
		
		public AdditionInputBody(AdditionInputBody value)
		{
			/// Initiliaze the protected variables
			m_AdditionInput = new AdditionInput();
			m_AdditionInput.setParent(this);
			
			/// Copy the values
			m_AdditionInput = value.m_AdditionInput;
			m_AdditionInput.setParent(this);
			/// This code is currently not supported
		}
		
		public void finalize()
		{
		}
		
	}
	protected Header m_Header;
	protected AdditionInputBody m_AdditionInputBody;
	public long getID()
	{
	return ID;
 }
	public QueryAddition.Header getHeader()
	{
		return m_Header;
	}
	
	public void setHeader(Header value)
	{
		m_Header = value;
	}
	
	public QueryAddition.AdditionInputBody getAdditionInputBody()
	{
		return m_AdditionInputBody;
	}
	
	public int setAdditionInputBody(AdditionInputBody value)
	{
		m_AdditionInputBody = value;
		return 0;
	}
	
	/**
	 * Returns the number of bytes the used data members of the class occupies
	 * in the buffer. This is not the number of bytes the data type occupies in 
	 * Java, but the bytes expected on the wire.
	 * 
	 * @return
	 */
	public long getSize()
	{
		int size = 0;
		
		size += m_Header.getSize();
		size += m_AdditionInputBody.getSize();
		
		return size;
	}
	
	public void encode(ByteBuffer bytes, int pos)
	{
		
		if (bytes.array() == null)
		{
			return;
		}
		if(bytes.order() != ByteOrder.LITTLE_ENDIAN)
		{
			bytes.order(ByteOrder.LITTLE_ENDIAN);
		}
		
		m_Header.encode(bytes, pos);
		pos += m_Header.getSize();
		m_AdditionInputBody.encode(bytes, pos);
		pos += m_AdditionInputBody.getSize();
	}
	
	public void decode(ByteBuffer bytes, int pos)
	{
		
		if (bytes.array() == null)
		{
			return;
		}
		if(bytes.order() != ByteOrder.LITTLE_ENDIAN)
		{
			bytes.order(ByteOrder.LITTLE_ENDIAN);
		}
		
		m_Header.decode(bytes, pos);
		pos += m_Header.getSize();
		m_AdditionInputBody.decode(bytes, pos);
		pos += m_AdditionInputBody.getSize();
	}
	
	public QueryAddition setAs(QueryAddition value)
	{
		m_Header = value.m_Header;
		m_AdditionInputBody = value.m_AdditionInputBody;
		
		return this;
	}
	
	public boolean isEqual(QueryAddition value)
	{
		if (!m_Header.isEqual(value.getHeader()))
		{
			return false;
		}
		if (!m_AdditionInputBody.isEqual(value.getAdditionInputBody()))
		{
			return false;
		}
		
		return true;
	}
	
	public boolean notEquals(QueryAddition value)
	{
		return !isEqual(value);
		}
		
	public QueryAddition()
	{
		m_Header = new Header();
		m_AdditionInputBody = new AdditionInputBody();
		m_IsCommand = false;
	}
	
	public  QueryAddition(QueryAddition value)
	{
		/// Initiliaze the protected variables
		m_Header = new Header();
		m_AdditionInputBody = new AdditionInputBody();
		m_IsCommand = false;
		
		/// Copy the values
		m_Header = value.m_Header;
		m_AdditionInputBody = value.m_AdditionInputBody;
	}
	
}
